pkg/api/customization/monitor/handler_utils.go

```diff
@@ -113,34 +177,108 @@ func (p *projectGraphInputParser) parseFilter() error {
-	Input              *v3.QueryGraphInput
-	UserContext        *config.UserContext
-	WorkloadController workload.CommonController
+func (p *projectGraphInputParser) parseProjectMetricParams(ctx context.Context, userContext *config.UserContext) error {
+	if p.Input.MetricParams == nil {
+		p.Input.MetricParams = make(map[string]string)
+		return nil
+	}
+
+	switch p.ResourceType {
+	case ResourceWorkload:
+		workloadName := p.Input.MetricParams["workloadName"]
+		workloadController := workload.NewWorkloadController(ctx, userContext.UserOnlyContext(), nil)
+		workload, err := workloadController.GetByWorkloadID(workloadName)
+		if err != nil {
+			return fmt.Errorf("get workload %s failed, %v", workloadName, err)
+		}
+		_, ns, _, err := parseWorkloadName(workloadName)
+		if err != nil {
+			return err
+		}
+		set := labels.Set(workload.SelectorLabels)
+
+		var podNames []string
+		pods, err := userContext.Core.Pods(ns).List(metav1.ListOptions{LabelSelector: set.String()})
+		if err != nil {
+			return fmt.Errorf("list pod failed, %v", err)
+		}
+		for _, pod := range pods.Items {
+			podNames = append(podNames, pod.Name)
+		}
+		p.Input.MetricParams["podName"] = strings.Join(podNames, "|")
+		p.Input.MetricParams["namespace"] = ns
+	case ResourcePod:
+		podName := p.Input.MetricParams["podName"]
+		if podName == "" {
+			return fmt.Errorf("pod name is empty")
+		}
+		ns, name := ref.Parse(podName)
+		p.Input.MetricParams["namespace"] = ns
+		p.Input.MetricParams["podName"] = name
+	case ResourceContainer:
+		podName := p.Input.MetricParams["podName"]
+		if podName == "" {
+			return fmt.Errorf("pod name is empty")
+		}
+		ns, name := ref.Parse(podName)
+		p.Input.MetricParams["namespace"] = ns
+		p.Input.MetricParams["podName"] = name
+
+		containerName := p.Input.MetricParams["containerName"]
+		if containerName == "" {
+			return fmt.Errorf("container name is empty")
+		}
+	}
+	return nil
+}
+
+type projectAuthChecker struct {
+	ProjectID   string
+	Input       *v3.QueryGraphInput
+	UserContext *config.UserContext
+	apiContext  *types.APIContext
 }
 
-func newAuthChecker(ctx context.Context, userContext *config.UserContext, input *v3.QueryGraphInput, projectID string) *authChecker {
-	return &authChecker{
-		ProjectID:          projectID,
-		Input:              input,
-		UserContext:        userContext,
-		WorkloadController: workload.NewWorkloadController(ctx, userContext.UserOnlyContext(), nil),
+func newProjectAuthChecker(userContext *config.UserContext, apiContext *types.APIContext, input *v3.QueryGraphInput, projectID string) *projectAuthChecker {
+	return &projectAuthChecker{
+		ProjectID:   projectID,
+		Input:       input,
+		UserContext: userContext,
+		apiContext:  apiContext,
 	}
 }
 
-func (a *authChecker) check() error {
+func (a *projectAuthChecker) check() error {
+	var project mgmtclient.Project
+	if err := access.ByID(a.apiContext, &managementschema.Version, mgmtclient.ProjectType, a.ProjectID, &project); err != nil {
```

Comment:
Same as `clusterAuthChecker`, maybe we need to use something like https://github.com/rancher/rancher/blob/master/pkg/api/customization/cluster/actionhandler.go#L52-L58 to check the privileges.